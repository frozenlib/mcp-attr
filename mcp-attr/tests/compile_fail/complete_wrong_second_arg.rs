use mcp_attr::{server::{mcp_server, McpServer, RequestContext}, Result};

struct TestServer;

async fn complete_wrong_second_arg(_value: &str, cx: RequestContext) -> Result<Vec<String>> {
    Ok(vec!["test".to_string()])
}

#[mcp_server]
impl McpServer for TestServer {
    #[prompt]
    async fn test_prompt(&self, #[complete(complete_wrong_second_arg)] msg: String) -> Result<String> {
        Ok(msg)
    }
}

fn main() {}