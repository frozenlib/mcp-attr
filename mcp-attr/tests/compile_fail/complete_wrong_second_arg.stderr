error[E0308]: mismatched types
 --> tests/compile_fail/complete_wrong_second_arg.rs:5:1
  |
5 | #[complete_fn]
  | ^^^^^^^^^^^^^^ expected `RequestContext`, found `&RequestContext`
6 | async fn complete_wrong_second_arg(_value: &str, cx: RequestContext) -> Result<Vec<String>> {
  |          ------------------------- arguments to this function are incorrect
  |
note: function defined here
 --> tests/compile_fail/complete_wrong_second_arg.rs:6:10
  |
6 | async fn complete_wrong_second_arg(_value: &str, cx: RequestContext) -> Result<Vec<String>> {
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^               ------------------
  = note: this error originates in the attribute macro `complete_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: unused variable: `cx`
 --> tests/compile_fail/complete_wrong_second_arg.rs:6:50
  |
6 | async fn complete_wrong_second_arg(_value: &str, cx: RequestContext) -> Result<Vec<String>> {
  |                                                  ^^ help: if this is intentional, prefix it with an underscore: `_cx`
  |
  = note: `#[warn(unused_variables)]` on by default
